stats/record.go

```diff
@@ -84,6 +93,23 @@ func RecordWithTags(ctx context.Context, mutators []tag.Mutator, ms ...Measureme
 	return RecordWithOptions(ctx, WithTags(mutators...), WithMeasurements(ms...))
 }
 
+// ResolveOptions determines the full set of Tags, Measurements, etc from the
+// provided Options and context.Context. This is mostly useful when using
+// multiple exporters.
+func ResolveOptions(ctx context.Context, ros ...Options) (*ResolvedOptions, error) {
```

Comment:
Not sure how this will be used. Can you provide an example?

Reply:
I want to be able to dispatch to different Meter interfaces based on the underlying resource. Unfortunately, the mechanism for this today is:

    stats.RecordWithOptions(ctx, stats.WithMeasurements(...), stats.WithTags(...), ...)

It's hard to extract the tags from the `stats.Options` because the mutation functions act on a `*createRecordOption`, which is non-exported and doesn't have an interface. Additionally, Meter is in the `view` package, while RecordWithOptions is in the `stats` package, so there's no good way to smuggle the data across the package boundary.

Reply:
This will make the Tags readable, which defeats the original purpose of it being only readable by the library.
Let me understand this. The flow is 
- Call the ResolveOptions()
- Determine resource from the Tags. Assuming here, that you need all tags (tags passed to ResolveOptions and tags present in the context prior to calling ResolveOptions).
- Determine the Meter/Exporter based on the resource.
- Record the value against the Meter.

Reply:
I'm not sure I understand -- is this about making the `stats.Options` not readable outside the library?

The other option (which may be what you're suggesting) is to either add a `WithMeter` option or a `RecordToMeter` method here which would use the logic internally.

Would you like me to take the `WithMeter` stats option for a spin? I hadn't considered it until you pushed a bit -- thanks!

Reply:
> Would you like me to take the WithMeter stats option for a spin? I hadn't considered it until you pushed a bit -- thanks!

Sure.

Reply:
Done. I think I like this better, but I had to re-export `RetrieveData` from the `Meter` interface for testing.